protocol/contracts/v2/TreasuryReservesVault.sol

```diff
@@ -80,395 +95,577 @@ contract TreasuryReservesVault is ITreasuryReservesVault, TempleElevatedAccess {
+        if (baseStrategyDepositThreshold < baseStrategyWithdrawalBuffer) revert CommonEventsAndErrors.InvalidParam();
+
+        emit BorrowTokenSet(address(token), baseStrategy, baseStrategyWithdrawalBuffer, baseStrategyDepositThreshold, dToken);
+        borrowTokens[token] = BorrowTokenConfig({
+            baseStrategy: ITempleBaseStrategy(baseStrategy),
+            baseStrategyWithdrawalBuffer: baseStrategyWithdrawalBuffer,
+            baseStrategyDepositThreshold: baseStrategyDepositThreshold,
+            dToken: ITempleDebtToken(dToken)
+        });
+        _borrowTokenSet.add(address(token));
+    }
 
-        emit BaseStrategySet(_baseStrategy);
+    /**
+     * @notice Remove the borrow token configuration. 
+     */
+    function removeBorrowToken(
+        IERC20 token
+    ) external override onlyElevatedAccess {
+        if (!_borrowTokenSet.contains(address(token))) revert BorrowTokenNotEnabled();
+        emit BorrowTokenRemoved(address(token));
+        delete borrowTokens[token];
+        _borrowTokenSet.remove(address(token));
     }
 
     /**
      * @notice Set the Treasury Price Index (TPI) Oracle
      */
     function setTpiOracle(address newTpiOracle) external override onlyElevatedAccess {
+        if (address(newTpiOracle) == address(0)) revert CommonEventsAndErrors.InvalidAddress();
         emit TpiOracleSet(newTpiOracle);
         tpiOracle = ITreasuryPriceIndexOracle(newTpiOracle);
     }
 
     /**
-     * @notice The Treasury Price Index - the target price of the Treasury, in `stableToken` terms.
+     * Pause all strategy borrow and repays
      */
-    function treasuryPriceIndex() public view override returns (uint256) {
-        return tpiOracle.treasuryPriceIndex();
+    function setGlobalPaused(bool _pauseBorrow, bool _pauseRepays) external override onlyElevatedAccess {
+        emit GlobalPausedSet(_pauseBorrow, _pauseRepays);
+        globalBorrowPaused = _pauseBorrow;
+        globalRepaysPaused = _pauseRepays;
     }
 
     /**
-     * @notice Track the deployed version of this contract. 
+     * Add a new strategy
      */
-    function apiVersion() external pure override returns (string memory) {
-        return API_VERSION;
+    function addStrategy(
+        address strategy, 
+        int256 underperformingEquityThreshold,
+        ITempleStrategy.AssetBalance[] calldata debtCeiling
+    ) external override onlyElevatedAccess {
+        if (_strategySet.contains(strategy)) revert AlreadyEnabled();
+        emit StrategyAdded(strategy, underperformingEquityThreshold);
+
+        StrategyConfig storage strategyConfig = strategies[strategy];
+        strategyConfig.underperformingEquityThreshold = underperformingEquityThreshold;
+
+        address _token;
+        uint256 _length = debtCeiling.length;
+        for (uint256 i; i < _length; ++i) {
+            _token = debtCeiling[i].asset;
+            strategyConfig.debtCeiling[IERC20(_token)] = debtCeiling[i].balance;
```

Comment:
nit: is ETH going to be wrapped to WETH?

Reply:
Yes correct - this is intentionally simplified to just use weth